fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -701,8 +709,8 @@ private Map<RecordType, Collection<IndexMaintainer>> getSyntheticMaintainers(@No
         final RecordMetaData metaData = getRecordMetaData();
         return syntheticRecordTypes.stream().map(metaData::getSyntheticRecordType).collect(Collectors.toMap(Function.identity(), syntheticRecordType -> {
             List<IndexMaintainer> indexMaintainers = new ArrayList<>();
-            syntheticRecordType.getIndexes().stream().map(this::getIndexMaintainer).forEach(indexMaintainers::add);
-            syntheticRecordType.getMultiTypeIndexes().stream().map(this::getIndexMaintainer).forEach(indexMaintainers::add);
+            syntheticRecordType.getIndexes().stream().filter(index -> !isIndexDisabled(index)).map(this::getIndexMaintainer).forEach(indexMaintainers::add);
```

Comment:
This got me thinking about other index states, do we need to call `updateWhileWriteOnly` if the index is in `writeOnly`

Reply:
Yes, we probably do want to call `updateWhileWriteOnly` in that case. For most maintainers, if the index is idempotent, that works out to be the same thing, but yeah, I think if you had a `COUNT` index (say) defined on a synthetic record type, then this won't correctly update the index while the index is being built.

That being said, I'm not _actually_ sure that `COUNT` indexes on synthetic record types can be built at all. In particular, I think it might actually double count any existing record, if I'm reading the logic in `IndexingBase` correctly:

https://github.com/FoundationDB/fdb-record-layer/blob/d4478d6d8c39760a096f7217c94f54a617adf0cb/fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingBase.java#L880-L897

And in any case, the logic to determine whether to index a record or not would have to be adjusted, essentially because synthetic record types aren't indexed in primary key order.

We may want to add `IndexValidation` logic to stop us from building non-idempotent indexes on synthetic types. That feels like it may belong in a separate PR. It may be worth switching `runSyntheticMaintainers` in this PR to call `update` or `updateWhileWriteOnly` as appropriate, though.

Reply:
Considering the time constraints on this, and current supported indexes (basically only lucene), I'd suggest that we file a GitHub issue on this and approach it at the same time we expand support for COUNT indexes over synthetic types